app/components/calendar-input.js

```diff
@@ -66,6 +68,16 @@ export default Ember.TextField.extend({
     }
   },
 
+  didReceiveAttrs() {
+    this._super(...arguments);
+    const enablePreviousDate = this.get("enablePreviousDate");
+    if (enablePreviousDate) {
+      PICKADATE_CONFIG.min = moment()
```

Comment:
an all caps variable is a convention for a constant. So modifying it is a big no no.

Especially given that this object is _shared_ across all `calendar-input` component instanciation. That means that if only one modifies it, all the others in the future will have that modifictation.

I would suggest having a constant for the default config, and then extending it

e.g

```javascript

const MIN_DATE = moment().subtract(2, "years").toDate();
const DEFAULT_PICKADATE_CONFIG = {
   // something
};

// This is just a helper to avoid doing the ternaries that you're doing in date-picker.js. Thought it was nicer
const optional = (val) => ({
   if: (cond) => cond ? val : undefined
});

// ...

{

  didReceiveAttrs() {
    this._super(...arguments);
    const enablePreviousDate = this.get("enablePreviousDate");
    this.set('datePickerConfig', {
       ...DEFAULT_PICKADATE_CONFIG,
       min: optional(MIN_DATE).if(enablePreviousDate)
    });

    // and then you use `datePickerConfig` inside the component
  }
}
```